doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst

```diff
@@ -8,16 +8,154 @@ NEP 35 — Array Creation Dispatching With __array_function__
 :Status: Draft
 :Type: Standards Track
 :Created: 2019-10-15
-:Updated: 2020-08-06
+:Updated: 2020-08-17
 :Resolution:
 
 Abstract
 --------
 
 We propose the introduction of a new keyword argument ``like=`` to all array
-creation functions to permit dispatching of such functions by the
-``__array_function__`` protocol, addressing one of the protocol shortcomings,
-as described by NEP-18 [1]_.
+creation functions, this argument permits the creation of an array based on
+a non-NumPy reference array passed via that argument, resulting in an array
+defined by the downstream library implementing that type, which also implements
+the ``__array_function__`` protocol. With this we address one of that
+protocol's shortcomings, as described by NEP 18 [1]_.
+
+Motivation and Scope
+--------------------
+
+Many are the libraries implementing the NumPy API, such as Dask for graph
+computing, CuPy for GPGPU computing, xarray for N-D labeled arrays, etc. All
+the libraries mentioned have yet another thing in common: they have also adopted
+the ``__array_function__`` protocol. The protocol defines a mechanism allowing a
```

Comment:
I would make this part as obvious as possible. Something along the lines of 

```
the ``__array_function__`` protocol; a protocol that allows NumPy to understand
how to treat downstream objects as if they are the native ndarray object. Hence
the community while using various libraries, still use a unified NumPy API. This 
not only brings great convenience for standardization but also removes the 
burden of relearning/recoding the same things for every other new object.
```

I am not sure about the exact wording but intention would in the ballpark.

Reply:
I did a few minor tweaks and added a comment that this mechanism is called a "dispatcher", just for the benefit of the rest of the proposal. Let me know if something is still unclear with the new phrasing.